librz/util/subprocess.c

```diff
@@ -927,6 +929,13 @@ RZ_API RzSubprocess *rz_subprocess_start_opt(RzSubprocessOpt *opt) {
 	int stdin_pipe[2] = { -1, -1 };
 	int stdout_pipe[2] = { -1, -1 };
 	int stderr_pipe[2] = { -1, -1 };
+
+	/* There seems to be no reasonable use case for having pipes when forking
+	using forkpty, hence we will bypass all the pipe creation logic */
+	if (opt->fork_mode == RZ_SUBPROCESS_FORKPTY) {
```

Comment:
instead of `goto`, extract the pipe handling in a separate function, like

```
if (opt->for_mode != RZ_SUBPROCESS_FORKPTY && !start_handle_pipes(proc, opt)) {
   goto error;
}
```